widgets/wifiiw.lua

```diff
@@ -0,0 +1,89 @@
+---------------------------------------------------
+-- Licensed under the GNU General Public License v2
+--  * (c) 2010, Adrian C. <anrxc@sysphere.org>
+---------------------------------------------------
+
+-- {{{ Grab environment
+local tonumber = tonumber
+local math = { ceil = math.ceil }
+local setmetatable = setmetatable
+local helpers = require("vicious.helpers")
+local io = {
+    open  = io.open,
+    popen = io.popen
+}
+local string = {
+    find  = string.find,
+    match = string.match
+}
+-- }}}
+
+
+-- Wifiiw: provides wireless information for a requested interface using iw instead of deprecated iwconfig
+-- vicious.widgets.wifiiw
+local wifiiw = {}
+
+
+-- {{{ Variable definitions
+local iw = "iw"
+local iwcpaths = { "/sbin", "/usr/sbin", "/usr/local/sbin", "/usr/bin" }
+-- }}}
+
+
+-- {{{ Wireless widget type
+local function worker(format, warg)
+    if not warg then return end
+
+    -- Default values
+    local winfo = {
```

Comment:
When I think about, we could also add an empty table here:

```
local winfo = {}
```

and then assign default values to fields like this:

```lua
winfo["{ssid}"] =  -- SSID can have almost anything in it until new line
    string.match(iwresult, 'SSID: ([^\n]*)') or "N/A"
```